doc/nrf/libraries/bootloader/bl_monotonic_counters.rst

```diff
@@ -0,0 +1,47 @@
+.. _doc_bl_monotonic_counters:
+
+Monotonic counters
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The monotonic counters library implements the logic for readind and writing uint16_t counters which can only increase.
+
+The library is used by :ref:`bootloader` to keep track of the version of the application.
+The monotonic counters are stored as part of the bootloader storage :ref:`doc_bl_storage`.
+When updating the application, the :ref:`doc_bl_validation` library checks if the new version is newer than the current version and only accepts the new image if it is.
+After updating, the application must then store the new version information in the bootloader storage.
+
+This functionality is implemented as a monotonic version counter that contains a series of 16-bit integer values.
+Each update to the counter is written to the next available slot.
+When reading the counter, the bootloader storage library iterates over each slot and returns the largest value.
+
+The number of available slots, thus the number of different version numbers that can be stored, is configurable through :kconfig:option:`CONFIG_SB_NUM_VER_COUNTER_SLOTS`.
+
+The monotonic counter is enabled by default.
+You can disable it through :kconfig:option:`CONFIG_SB_MONOTONIC_COUNTER`.
+If the counter is enabled, the :ref:`doc_bl_validation` library checks it against an image's version during :c:func:`bl_validate_firmware`.
+
+See :ref:`bootloader_provisioning` for more information about the provisioned data and how the bootloader uses it.
+
+This library is also used by :ref:`MCUboot <mcuboot:mcuboot_ncs>` bootloader when the hardware downgrade prevention option is enabled.
+The option is disabled by default and it can be enabled by setting :kconfig:option:`CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION`.
+When both bootloaders :ref:`bootloader` and :ref:`MCUboot <mcuboot:mcuboot_ncs>` are enabled the separate monotonic counters will be generated for each of them.
+
+The number of available slots for :ref:`MCUboot <mcuboot:mcuboot_ncs>` is configurable through :kconfig:option:`MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_SLOTS`.
+:ref:`MCUboot <mcuboot:mcuboot_ncs>` is using one monotonic counter for each updateable image(excluding the image of :ref:`MCUboot <mcuboot:mcuboot_ncs>` itself since it's version is verified by :ref:`bootloader`).
+
+You can find tests for the library at :file:`tests/subsys/bootloader/bl_monotonic_counters/`.
+
```

Comment:
I think we should align with the template and have a "Samples using the library" section. Especially since it's not expected to be used by "normal" NCS applications.

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/templates/library_template_README.html#samples-using-the-library

Reply:
We can move some of the docs here to this section.